src/openms/include/OpenMS/FORMAT/IdXMLFile.h

```diff
@@ -115,7 +114,7 @@ namespace OpenMS
 
     /// Add data from ProteinGroups to a MetaInfoInterface
     void addProteinGroups_(MetaInfoInterface& meta, const std::vector<ProteinIdentification::ProteinGroup>& groups,
-                           const String& group_name, const std::unordered_map<std::string, UInt>& accession_to_id);
+                           const String& group_name, const std::unordered_map<std::string, UInt>& accession_to_id, XMLHandler::ActionMode mode);
```

Comment:
what does the action mode do again?

Reply:
IIRC It can be read or write and I think in this case is only used to generate a warning string that contains Error during write or Error during read depending on this.